numpy/f2py/crackfortran.py

```diff
@@ -2433,7 +2433,7 @@ def get_parameters(vars, global_params={}):
                     v = v.replace(*repl)
             v = kind_re.sub(r'kind("\1")', v)
             v = selected_int_kind_re.sub(r'selected_int_kind(\1)', v)
-            if isinteger(vars[n]) and not selected_kind_re.match(v):
+            if not selected_kind_re.match(v):
```

Comment:
Let me point out that currently the above fix won't work on complex... I just saw further down the comment on the complex stuff. I will try and fix this as well. hold on.